gen_vlen.py

```diff
@@ -0,0 +1,43 @@
+import sys
+import os
+import math
+
+vlen = int(sys.argv[1])
+elen = int(sys.argv[2])
+sail_path = 'model/riscv_vlen.sail'
+
+# Check value of VLEN
+if vlen < 128:
+    vlen = 128
+    print('WARN: VLEN less than 128; setting VLEN to 128\n')
+if vlen % 2 != 0:
+    vlen = vlen - (vlen % 2)
+    print('WARN: VLEN not a power of 2; setting VLEN to %d\n' % vlen)
+
+# Check value of ELEN
+if elen < 8:
+    elen = 8
+    print('WARN: ELEN less than 8; setting ELEN to 8\n')
+if elen % 2 != 0:
+    elen = elen - (elen % 2)
+    print('WARN: ELEN not a power of 2; setting ELEN to %d\n' % elen)
+
+if os.path.exists(sail_path):
+    os.remove(sail_path)
+
+lb = ['/* Define the VLEN value for the architecture */\n',
+      '\n',
+      'type vlen          : Int = %d\n' % vlen,
+      'type vlen_bytes    : Int = %d\n' % (vlen / 8),
+      'type vstart_int    : Int = %d\n' % math.log(vlen, 2),
+      'type vstartbits          = bits(vstart_int)\n',
+      '\n',
+      'type elen          : Int = %d\n' % elen,
+      '\n']
+
+    
+fh_sail = open('model/riscv_vlen.sail', 'w')
+fh_sail.writelines(lb)
+fh_sail.close()
+
+exit()
```

Comment:
This opens the same discussion about configurability and the idea of auto-generating SAIL code. While this script is trivial, and one can clearly reason its inputs and outputs, I wonder if it's better to have these as options to the simulators cmd line args instead of generating them like this, Is there anything that prevents us from assigning these at runtime?

Reply:
Personally, I think this is not needed, if we cannot set these at runtime for whatever reason, then people can just change them in that file, perhaps we should be checking that VLEN and ELEN  actually make sense inside the model rather than here ?

Reply:
I think we are going to prohibit any dynamic change of the Sail model. The Sail model is a constant, and the C-model it generates is a constant. Any behavioral change will be parameters that are passed to the Csim as commandline arguments (some of them rather complex, which is what we are trying to pin downs, e.g. lists of lists - but these are simple)